geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/list/AbstractRPopIntegrationTest.java

```diff
@@ -0,0 +1,155 @@
+import static org.apache.geode.redis.internal.RedisConstants.ERROR_WRONG_TYPE;
+import static org.apache.geode.test.dunit.rules.RedisClusterStartupRule.BIND_ADDRESS;
+import static org.apache.geode.test.dunit.rules.RedisClusterStartupRule.REDIS_CLIENT_TIMEOUT;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import redis.clients.jedis.HostAndPort;
+import redis.clients.jedis.JedisCluster;
+import redis.clients.jedis.Protocol;
+
+import org.apache.geode.redis.ConcurrentLoopingThreads;
+import org.apache.geode.redis.RedisIntegrationTest;
+
+public abstract class AbstractRPopIntegrationTest implements RedisIntegrationTest {
+  public static final String KEY = "key";
+  public static final String PREEXISTING_VALUE = "preexistingValue";
+  protected JedisCluster jedis;
+
+  @Before
+  public void setUp() {
+    jedis = new JedisCluster(new HostAndPort(BIND_ADDRESS, getPort()), REDIS_CLIENT_TIMEOUT);
+  }
+
+  @After
+  public void tearDown() {
+    flushAll();
+    jedis.close();
+  }
+
+  // Overridden in RPopIntegrationTest until we implement Redis 6.2+ semantics
+  @Test
+  public void rpop_givenWrongNumOfArgs_returnsError() {
+    assertAtMostNArgs(jedis, Protocol.Command.RPOP, 2);
+  }
+
+  @Test
+  public void rpop_withNonListKey_Fails() {
+    jedis.set("nonKey", PREEXISTING_VALUE);
+    assertThatThrownBy(() -> jedis.rpop("nonKey")).hasMessage(ERROR_WRONG_TYPE);
+  }
+
+  @Test
+  public void rpop_withNonExistentKey_returnsNull() {
+    assertThat(jedis.rpop("nonexistentKey")).isNull();
+  }
+
+  @Test
+  public void rpop_returnsRightmostMember() {
+    jedis.lpush(KEY, "e1", "e2");
+    String result = jedis.rpop(KEY);
+    assertThat(result).isEqualTo("e1");
+  }
+
+  @Test
+  public void rpop_removesKey_whenLastElementRemoved() {
+    final String keyWithTagForKeysCommand = "{tag}" + KEY;
+
+    jedis.lpush(keyWithTagForKeysCommand, "e1");
+    jedis.rpop(keyWithTagForKeysCommand);
+    assertThat(jedis.keys(keyWithTagForKeysCommand)).isEmpty();
+  }
+
+  @Test
+  public void rpop_removesKey_whenLastElementRemoved_multipleTimes() {
+    final String key = KEY;
+
+    jedis.lpush(key, "e1");
+    assertThat(jedis.rpop(key)).isEqualTo("e1");
+    assertThat(jedis.rpop(key)).isNull();
+    assertThat(jedis.rpop(key)).isNull();
+    assertThat(jedis.exists(key)).isFalse();
+  }
+
+  @Test
+  public void rpop_removesElementsInCorrectOrder_onRepeatedInvocation() {
+    jedis.lpush(KEY, "e1");
+    jedis.lpush(KEY, "e2");
+    jedis.lpush(KEY, "e3");
+
+    String result = jedis.rpop(KEY);
+    assertThat(result).isEqualTo("e1");
+
+    result = jedis.rpop(KEY);
+    assertThat(result).isEqualTo("e2");
+
+    result = jedis.rpop(KEY);
+    assertThat(result).isEqualTo("e3");
```

Comment:
This can be simplified slightly to:
```
    assertThat(jedis.rpop(KEY)).isEqualTo("e1");
    assertThat(jedis.rpop(KEY)).isEqualTo("e2");
    assertThat(jedis.rpop(KEY)).isEqualTo("e3");
```